server/game/GameActions/AttachAction.ts

```diff
@@ -28,6 +35,8 @@ export class AttachAction extends CardGameAction {
             return false;
         } else if(!properties.attachment || properties.attachment.anotherUniqueInPlay(context.player) || !properties.attachment.canAttach(card, context)) {
```

Comment:
I think `anotherUniqueInPlay` only needs checking if we're taking control?

Reply:
Tests for Gunbai fail unless we do this as it needs to check if another unique is in play but there is no "taking control"